### PR TITLE
Connection clean up and upgrade httpClient

### PR DESCRIPTION
**Pull request by jjangsfdc**
Upgrade Apache httpClient lib from 4.2.1 to 4.5.2 and update depreciated API calls with connection cleanups.
W-3238933

### CODE REVIEW

File: src/main/java/com/salesforce/dataloader/client/DefaultSimplePost.java [hunk 1]
```diff
@@ -72,62 +73,80 @@
...
-            client.getCredentialsProvider().setCredentials(scope, credentials);
-        }
-
+        CloseableHttpClient client = HttpClients.createDefault();
         try {
-            if (ntlmDomain.length() > 0) {
-                // need to send a HEAD request to trigger NTLM authentication
-                HttpHead head = new HttpHead("http://salesforce.com");
-                client.execute(head);
-                head.releaseConnection();
```
Comment (rjmazzeo) [inline src/main/java/com/salesforce/dataloader/client/DefaultSimplePost.java]: We don't need to issue a head request for nt proxy servers anymore?

Comment (rjmazzeo) [inline src/main/java/com/salesforce/dataloader/client/DefaultSimplePost.java]: never mind... I seeit.

File: src/main/java/com/salesforce/dataloader/client/HttpClientTransport.java [hunk 2]
```diff
@@ -90,64 +95,83 @@ public OutputStream connect(String url, String soapAction) throws IOException {
...
+                }
+
+            } finally {
+                if (response != null)
+                    response.close();
             }
-   
         } finally {
-            post.releaseConnection();
+            client.close();
```
Comment (xbiansf) [inline src/main/java/com/salesforce/dataloader/client/HttpClientTransport.java]: Looks good.  Like the new way of closing connection.

Comment (jjangsfdc) [inline src/main/java/com/salesforce/dataloader/client/HttpClientTransport.java]: :) Sorry the diffs are kind of hard to follow. Adding Try block messed around it a lot!

**Comment by rjmazzeo** [comment]
I think it looks good enough to try. just see what happens with intergration tests.

We should also need to vet the  proxy changes

**Comment by jjangsfdc** [comment]
I've already ran the integration tests and all passed.

**Comment by apoorvashenoy** [comment]
I would test OAuth as well. But Rocky's tests should cover those.

**Comment by apoorvashenoy** [comment]
Not that I am aware of.
